src/coreclr/gc/gc.cpp

```diff
@@ -28134,7 +28134,7 @@ void gc_heap::thread_final_regions (bool compact_p)
         heap_segment* current_region = heap_segment_rw (generation_start_segment (generation_of (gen_idx)));
         dprintf (REGIONS_LOG, ("gen%d start from %Ix", gen_idx, heap_segment_mem (current_region)));
 
-        while (current_region = find_first_valid_region (current_region, compact_p))
+        while (current_region == find_first_valid_region (current_region, compact_p))
```

Comment:
with this change it appears that some of the infinite loops within tests have now resulted in asserts, and no of failures are more. We should still take this change however so as to start fixing all these failures.

Reply:
if there's an infinite loop, we should debug that. but this is supposed to be =, not ==.

Reply:
huh.. should have looked closer. I assumed the build break was due to the operator. Will fix.